### DISCUSSION TITLE
[coding help]

### DISCUSSION THREAD

**Discussion by amantiwari-18** [mdn/mdn-community#656] [Content]
/C:/Users/noobg/AppData/Local/Pub/Cache/hosted/pub.dev/amplify_secure_storage-0.4.2/lib/src/amplify_secure_storage.vm.dart:138:27: Error: The method 'getApplicationSupportDirectory' isn't defined for the class 'AmplifySecureStorage'.

'AmplifySecureStorage' is from 'package:amplify_secure_storage/src/amplify_secure_storage.vm.dart' ('/C:/Users/noobg/AppData/Local/Pub/Cache/hosted/pub.dev/amplify_secure_storage-0.4.2/lib/src/amplify_secure_storage.vm.dart').
Try correcting the name to the name of an existing method, or defining a method named 'getApplicationSupportDirectory'.
final path = (await getApplicationSupportDirectory()).path;
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/C:/Users/noobg/AppData/Local/Pub/Cache/hosted/pub.dev/amplify_secure_storage-0.4.2/lib/src/path_provider_local.dart:31:11: Error: Method not found: 'MissingPlatformDirectoryException'.
throw MissingPlatformDirectoryException(
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
![Capture](https://github.com/mdn/mdn-community/assets/102050032/f0743831-131e-4a49-bfe6-b1f43d8e6092)

**Comment by bsmth** [comment]
MDN GitHub Discussions should be topics related to the MDN Web Docs project. If you have [general coding questions](https://developer.mozilla.org/en-US/docs/MDN/Community#general_support), you can try [Stack Overflow](https://stackoverflow.com/questions/) or other forums that can help you troubleshoot. I'm going to close this one for now. Thank you :)
